source/common/upstream/lazy_loader_impl.cc

```diff
@@ -0,0 +1,68 @@
+#include "common/upstream/lazy_loader_impl.h"
+
+#include "envoy/config/bootstrap/v2/bootstrap.pb.h"
+#include "envoy/upstream/lazy_loader.h"
+#include "envoy/event/dispatcher.h"
+
+#include "common/config/subscription_factory.h"
+#include "common/config/utility.h"
+#include "common/config/grpc_mux_impl.h"
+
+
+namespace Envoy {
+namespace Upstream {
+
+
+LazyLoaderImpl::LazyLoaderImpl(const envoy::config::bootstrap::v2::Bootstrap& bootstrap,
+                               const LocalInfo::LocalInfo& local_info,
+                               Event::Dispatcher& primary_dispatcher,
+                               Runtime::RandomGenerator& random,
+                               ClusterManagerImpl& cm,
+                               Stats::Store& stats) :
+                                 primary_dispatcher_(primary_dispatcher),
+                                 cluster_manager_(cm) {
+  ASSERT(bootstrap.dynamic_resources().has_cds_config());
+  if (bootstrap.dynamic_resources().has_ads_config()) {
+    // Create an ADS stream for the lazy loader. We can not reuse the cluster manager ADS stream.
```

Comment:
I'm not sure if this is what we want for ADS. This will become clearer once we resolve https://github.com/envoyproxy/data-plane-api/pull/524.